### DISCUSSION TITLE
Given a short address, convert it or obtain a long one (f0 -> f4)

### DISCUSSION THREAD

**Discussion by GildedHonour** [filecoin-project/lotus#10898] [General Q&A]
Given a short address f0, can it be converted into a long one -- f4? Preferably, locally without any request. Otherwise, via a RPC request to a node. Otherwise,via a third-party gateway API.

**Comment by TippyFlitsUK** [comment]
https://filecoinproject.slack.com/archives/CPFTWMY7N/p1684520861093529

**Comment by GildedHonour** [comment]
Who had marked it as the answer before I even looked at it?

**Reply by TippyFlitsUK** [reply]
You asked the same question in Slack which was then answered @GildedHonour.

**Comment by TippyFlitsUK** [comment]
To quote:

```
use StateAccountKey
But it only works for normal F1 F3 addresses
if you want lookup actor
you can use StateLookupRobustAddress
```

**Reply by TippyFlitsUK** [reply]
It is not possible to convert directly to f4 as stated in the linked thread!

**Reply by TippyFlitsUK** [reply]
Also - https://filecoinproject.slack.com/archives/CRK2LKYHW/p1687432664411409
